### ISSUE TITLE
KT-25722 - Android Studio IDE error when changing Gradle Kotlin DSL script dependencies

### ISSUE THREAD

**Issue by eskatos** [gradle/kotlin-dsl#980]
### Expected Behavior
No IDE error.

### Current Behavior

When editing a `*.gradle.kts` file, the following exception is thrown:

<details>
 <summary>LazyLightClassMemberMatchingError stack trace</summary>

```
Matched ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 0 parameters) to ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 1 parameters)
org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassMemberMatchingError$WrongMatch: Matched ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 0 parameters) to ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 1 parameters)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder.assertMatches(LazyLightClassDataHolder.kt:150)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder.access$assertMatches(LazyLightClassDataHolder.kt:32)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder$LazyLightClassData$getOwnMethods$$inlined$map$lambda$1.invoke(LazyLightClassDataHolder.kt:138)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder$LazyLightClassData$getOwnMethods$$inlined$map$lambda$1.invoke(LazyLightClassDataHolder.kt:100)
	at kotlin.SafePublicationLazyImpl.getValue(LazyJVM.kt:107)
	at org.jetbrains.kotlin.asJava.elements.KtLightMemberImpl.getClsDelegate(KtLightMemberImpl.kt)
	at org.jetbrains.kotlin.asJava.elements.KtLightMethodImpl$paramsList$2$1.invoke(KtLightMethodImpl.kt:54)
	at org.jetbrains.kotlin.asJava.elements.KtLightMethodImpl$paramsList$2$1.invoke(KtLightMethodImpl.kt:39)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList$_parameters$2.invoke(KtLightParameterList.kt:36)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList$_parameters$2.invoke(KtLightParameterList.kt:27)
	at kotlin.SafePublicationLazyImpl.getValue(LazyJVM.kt:107)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList.get_parameters(KtLightParameterList.kt)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList.getParameters(KtLightParameterList.kt:37)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod$uastParameters$2.invoke(KotlinUMethod.kt:71)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod$uastParameters$2.invoke(KotlinUMethod.kt:35)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod.getUastParameters(KotlinUMethod.kt)
	at org.jetbrains.uast.UAnnotationMethod$DefaultImpls.accept(UMethod.kt:101)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod.accept(KotlinUMethod.kt:35)
	at org.jetbrains.uast.internal.ImplementationUtilsKt.acceptList(implementationUtils.kt:23)
	at org.jetbrains.uast.UClass$DefaultImpls.accept(UClass.kt:63)
	at org.jetbrains.uast.kotlin.AbstractKotlinUClass.accept(KotlinUClass.kt:33)
	at org.jetbrains.uast.internal.ImplementationUtilsKt.acceptList(implementationUtils.kt:23)
	at org.jetbrains.uast.UFile$DefaultImpls.accept(UFile.kt:91)
	at org.jetbrains.uast.kotlin.KotlinUFile.accept(KotlinUFile.kt:30)
	at org.jetbrains.kotlin.android.folding.ResourceFoldingBuilder.buildFoldRegions(ResourceFoldingBuilder.kt:80)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:90)
	at com.intellij.lang.folding.CompositeFoldingBuilder.buildFoldRegions(CompositeFoldingBuilder.java:63)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:90)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:238)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:224)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getUpdateResult(FoldingUpdate.java:109)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.lambda$updateFoldRegions$0(FoldingUpdate.java:98)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.java:45)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:60)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.java:35)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:96)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.updateFoldRegions(FoldingUpdate.java:94)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegions(CodeFoldingManagerImpl.java:319)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegionsAsync(CodeFoldingManagerImpl.java:305)
	at com.intellij.codeInsight.daemon.impl.CodeFoldingPass.doCollectInformation(CodeFoldingPass.java:42)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:437)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1130)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:430)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:429)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:405)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:143)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:229)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:403)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:170)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)


org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassMemberMatchingError$WrongMatch: Matched ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 0 parameters) to ClsMethodImpl:Build_gradle MemberIndex(index=0) (with 1 parameters)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder.assertMatches(LazyLightClassDataHolder.kt:150)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder.access$assertMatches(LazyLightClassDataHolder.kt:32)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder$LazyLightClassData$getOwnMethods$$inlined$map$lambda$1.invoke(LazyLightClassDataHolder.kt:138)
	at org.jetbrains.kotlin.idea.caches.lightClasses.LazyLightClassDataHolder$LazyLightClassData$getOwnMethods$$inlined$map$lambda$1.invoke(LazyLightClassDataHolder.kt:100)
	at kotlin.SafePublicationLazyImpl.getValue(LazyJVM.kt:107)
	at org.jetbrains.kotlin.asJava.elements.KtLightMemberImpl.getClsDelegate(KtLightMemberImpl.kt)
	at org.jetbrains.kotlin.asJava.elements.KtLightMethodImpl$paramsList$2$1.invoke(KtLightMethodImpl.kt:54)
	at org.jetbrains.kotlin.asJava.elements.KtLightMethodImpl$paramsList$2$1.invoke(KtLightMethodImpl.kt:39)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList$_parameters$2.invoke(KtLightParameterList.kt:36)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList$_parameters$2.invoke(KtLightParameterList.kt:27)
	at kotlin.SafePublicationLazyImpl.getValue(LazyJVM.kt:107)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList.get_parameters(KtLightParameterList.kt)
	at org.jetbrains.kotlin.asJava.elements.KtLightParameterList.getParameters(KtLightParameterList.kt:37)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod$uastParameters$2.invoke(KotlinUMethod.kt:71)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod$uastParameters$2.invoke(KotlinUMethod.kt:35)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod.getUastParameters(KotlinUMethod.kt)
	at org.jetbrains.uast.UAnnotationMethod$DefaultImpls.accept(UMethod.kt:101)
	at org.jetbrains.uast.kotlin.declarations.KotlinUMethod.accept(KotlinUMethod.kt:35)
	at org.jetbrains.uast.internal.ImplementationUtilsKt.acceptList(implementationUtils.kt:23)
	at org.jetbrains.uast.UClass$DefaultImpls.accept(UClass.kt:63)
	at org.jetbrains.uast.kotlin.AbstractKotlinUClass.accept(KotlinUClass.kt:33)
	at org.jetbrains.uast.internal.ImplementationUtilsKt.acceptList(implementationUtils.kt:23)
	at org.jetbrains.uast.UFile$DefaultImpls.accept(UFile.kt:91)
	at org.jetbrains.uast.kotlin.KotlinUFile.accept(KotlinUFile.kt:30)
	at org.jetbrains.kotlin.android.folding.ResourceFoldingBuilder.buildFoldRegions(ResourceFoldingBuilder.kt:80)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:90)
	at com.intellij.lang.folding.CompositeFoldingBuilder.buildFoldRegions(CompositeFoldingBuilder.java:63)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:90)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:238)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:224)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getUpdateResult(FoldingUpdate.java:109)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.lambda$updateFoldRegions$0(FoldingUpdate.java:98)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.java:45)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:60)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.java:35)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:96)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.updateFoldRegions(FoldingUpdate.java:94)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegions(CodeFoldingManagerImpl.java:319)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.updateFoldRegionsAsync(CodeFoldingManagerImpl.java:305)
	at com.intellij.codeInsight.daemon.impl.CodeFoldingPass.doCollectInformation(CodeFoldingPass.java:42)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:437)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1130)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:430)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:429)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:405)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:143)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:229)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:403)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:170)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
</details>

.

![android-studio-exception](https://user-images.githubusercontent.com/132773/43144689-03436d9c-8f5e-11e8-80ca-24e62d49de43.gif)

### Your Environment
```
Android Studio 3.1.3
Build #AI-173.4819257, built on June 4, 2018
JRE: 1.8.0_152-release-1024-b01 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.13.4
```
With the Android Studio Kotlin Plugin `1.2.60-eap-44`

### Vendor issue

https://youtrack.jetbrains.com/issue/KT-25722

**Comment by NataliaUkhorskaya** [comment]
Issue in Kotlin Youtrack: https://youtrack.jetbrains.com/issue/KT-25722

**Comment by eskatos** [comment]
Fixed in 1.2.60
